src/gen_rpc_acceptor.erl

```diff
@@ -321,25 +329,39 @@ check_module_version_compat({M, Version}) ->
 check_module_version_compat(M) ->
     {true, M}.
 
-handle_cast({cast, M, F, A}, #state{socket=Socket, driver=Driver, peer=Peer, control=Control, list=List}) ->
+handle_cast(M, F, A, Ordered, #state{socket=Socket, driver=Driver, peer=Peer, control=Control, list=List}) ->
     {ModVsnAllowed, RealM} = check_module_version_compat(M),
     case check_if_module_allowed(RealM, Control, List) of
         true ->
             case ModVsnAllowed of
                 true ->
-                    ?log(debug, "event=cast_received driver=~s socket=\"~s\" peer=\"~s\" module=~s function=~s args=\"~0p\"", [Driver, gen_rpc_helper:socket_to_string(Socket), gen_rpc_helper:peer_to_string(Peer), RealM, F, A]),
-                    _Pid = erlang:spawn(RealM, F, A);
+                    ?tp(debug, gen_rpc_exec_cast,
+                        #{ mfa    => {M, F, A}
+                         , socket => gen_rpc_helper:socket_to_string(Socket)
+                         , peer   => gen_rpc_helper:peer_to_string(Peer)
+                         }),
+                    exec_cast(RealM, F, A, Ordered);
                 false ->
                     ?log(debug, "event=incompatible_module_version driver=~s socket=\"~s\" module=~s",[Driver, gen_rpc_helper:socket_to_string(Socket), RealM])
             end;
         false ->
             ?log(debug, "event=request_not_allowed driver=~s socket=\"~s\" control=~s method=cast module=~s",[Driver, gen_rpc_helper:socket_to_string(Socket), Control, RealM])
-    end;
-handle_cast(UnknownReq, #state{socket=Socket, driver=Driver, peer=Peer}) ->
-    ?log(debug, "event=invalid_cast_req driver=~s socket=\"~s\" peer=\"~s\" req=\"~p\"",
-         [Driver, gen_rpc_helper:socket_to_string(Socket),
-          gen_rpc_helper:peer_to_string(Peer), UnknownReq]),
-    error(invalid_cast_req).
+    end.
+
+exec_cast(M, F, A, _Ordered = true) ->
+    try apply(M, F, A)
+    catch
```

Comment:
It may actually make sense to run this function in a helper process to avoid GC and problems related with a long message queue.